extensions/BOP/twitch.js

```diff
@@ -0,0 +1,324 @@
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+              size: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "20",
+              },
+              font: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "font",
+              },
+            },
+          },
+          {
+            opcode: "image",
+            text: "create image with image [image] with width [width] with height [height] with position [position] is bottom [bottom] is top [top]",
+            blockType: Scratch.BlockType.COMMAND,
+            arguments: {
+              image: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "https://tekinical.github.io/twitchURI.png",
+              },
+              width: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "50",
+              },
+              height: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "50",
+              },
+              position: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "position",
+              },
+              bottom: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+              top: {
+                type: Scratch.ArgumentType.BOOLEAN,
+                defaultValue: false,
+              },
+            },
+          },
+          {
+            opcode: "closetext",
+            text: "close text",
+            blockType: Scratch.BlockType.COMMAND,
+          },
+          {
+            opcode: "closeimage",
+            text: "close image",
+            blockType: Scratch.BlockType.COMMAND,
+          },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Utilities",
+          },
+          {
+            opcode: "true",
+            text: "true",
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+          {
+            opcode: "false",
+            text: "false",
+            blockType: Scratch.BlockType.BOOLEAN,
+          },
+        ],
+        menus: {
+          font: {
+            acceptReporters: false,
+            items: "getFonts",
+          },
+          position: {
+            acceptReporters: false,
+            items: "getPositions",
+          },
+        },
+      };
+    }
+
+    setupchat(args) {
+      if (chat == null) {
+        var Link =
+          "https://nightdev.com/hosted/obschat/?theme=bttv_dark&channel=" +
+          Scratch.Cast.toString(args.username) +
+          "&fade=" +
+          Scratch.Cast.toNumber(args.fade) +
+          "&bot_activity=" +
+          Scratch.Cast.toString(args.bots) +
+          "&prevent_clipping=" +
+          Scratch.Cast.toString(args.prevent);
+        chat = document.createElement("iframe");
+        chat.style.width = args.width;
```

Comment:
Do you need to make sure to cast these too? Can it deal with other types?

Ask that question every time you're dealing with arguments.

Reply:
Casted every arguement.